scripts/cleanup_old_vms.py

```diff
@@ -44,167 +35,181 @@ def parse_cmd_line():
+    # Nested exceptions to try and be safe about the scanned values and to get complete results
+    failure = False
+    messages = []
+    status = NULL
     try:
-        providers_data = cfme_data.get("management_systems", {})
-        yaml_hosts = providers_data[provider_key]['hosts']
-        # pull host name and credentials from yaml, may be multiple hosts defined
-        hostname, host_cred_key = None, None  # Init in case not found in yaml
-        for host in yaml_hosts:
-            # Some of the vsphere environments are configured with IP as the host's name
-            if host_name in [host['name'], host.get('ipaddress')]:
-                hostname, host_cred_key = host['name'], host['credentials']
-                break
-
-        if not hostname or not host_cred_key:
-            raise KeyError('Could not find host {} and credentials for {} in yaml'
-                           .format(host_name, provider_key))
-        connect_kwargs = {
-            'username': credentials[host_cred_key]['username'],
-            'password': credentials[host_cred_key]['password'],
-            'hostname': hostname
-        }
-        datastore_path = re.findall(r'([^ds:`/*].*)', str(datastore_url))
-        command = 'find ~/{} -exec date -r {} \; -name {}.vmx | xargs  date -r'.format(
-            pathjoin(datastore_path[0], vm_name),
-            vm_name)
-
-        with SSHClient(**connect_kwargs) as ssh_client:
-            result = ssh_client.run_command(command)
-        parsed_time = parser.parse(str(result).rstrip())
-        return parsed_time.astimezone(pytz.timezone(str(get_localzone()))).replace(tzinfo=None)
-    except Exception as e:
-        logger.error(e)
-        return False
+        # Localize to UTC
+        vm_creation_time = provider_mgmt.vm_creation_time(vm_name)
+    except Exception:
+        failure = True
+        messages.append('{}: Exception getting creation time for {}'.format(provider_key, vm_name))
+        # This VM must have some problem, include in report even though we can't delete
+        try:
+            status = provider_mgmt.vm_status(vm_name)
+        except Exception:
+            failure = True
+            messages.append('{}: Exception getting status for {}'.format(provider_key, vm_name))
+            status = NULL
+    finally:
+        if failure:
+            for message in messages:
+                logger.exception(message)
+                print(message)
+            scan_failure_queue.put([VmReport(provider_key, vm_name, FAIL, status, NULL)])
+            return
 
+    # None of this should raise exceptions
+    vm_delta = now - vm_creation_time
+    print('{}: VM {} age: {}'.format(provider_key, vm_name, vm_delta))
+    data = VmData(provider_key, vm_name, str(vm_delta))
 
-def process_provider_vms(provider_key, matchers, delta, vms_to_delete):
+    # test age to determine which queue it goes in
+    queue = match_queue if delta < vm_delta else non_match_queue
+    queue.put([data])
```

Comment:
not sure I got, you use extend on get(). What interferes you to use append on get and item instead of list here ?